TestMiniController_FailMarshalKey_OnStart segfaults

https://github.com/dedis/dela/pull/246#issuecomment-1479948341

Reproduced on Windows:
```
--- FAIL: TestMiniController_FailMarshalKey_OnStart (0.15s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x38 pc=0xfbbc23]

goroutine 33 [running]:
testing.tRunner.func1.2({0x1132fa0, 0x17aa660})
        C:/Program Files/Go/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
        C:/Program Files/Go/src/testing/testing.go:1529 +0x39f
panic({0x1132fa0, 0x17aa660})
        C:/Program Files/Go/src/runtime/panic.go:884 +0x213
go.dedis.ch/dela/mino/minogrpc/certs.(*DiskStore).Store.func1({0x131e140?, 0xc0001ea018?})
        G:/projects/dedis/dela/mino/minogrpc/certs/disk.go:59 +0xc3
go.dedis.ch/dela/internal/testing/fake.(*InMemoryDB).Update(0x1ada1c60598?, 0xc000216940)
        G:/projects/dedis/dela/internal/testing/fake/store.go:105 +0x67
go.dedis.ch/dela/mino/minogrpc/certs.(*DiskStore).Store(0xc000216840, {0x131e230, 0xc0001ea000}, {0xc0000ee6e0, 0x15b, 0x15b})
        G:/projects/dedis/dela/mino/minogrpc/certs/disk.go:53 +0x15e
go.dedis.ch/dela/mino/minogrpc.(*overlay).makeCertificate(0xc0002300c0)
        G:/projects/dedis/dela/mino/minogrpc/server.go:627 +0x4d3
go.dedis.ch/dela/mino/minogrpc.newOverlay(0xc00022e0a0)
        G:/projects/dedis/dela/mino/minogrpc/server.go:504 +0x793
go.dedis.ch/dela/mino/minogrpc.NewMinogrpc({0x131b4c8, 0xc000212f30}, 0x0, {0x131f090, 0xc000212f90}, {0xc0003f08f0, 0x2, 0xc000204078?})
        G:/projects/dedis/dela/mino/minogrpc/mod.go:209 +0x4d5
go.dedis.ch/dela/mino/minogrpc/controller.miniController.OnStart({{0x1317100?, 0x1819430?}, {0x1321150?, 0xc0003f0860?}}, {0x1321198, 0xc000216800}, {0x131ab18, 0xc000204078})
        G:/projects/dedis/dela/mino/minogrpc/controller/mod.go:189 +0x505
go.dedis.ch/dela/mino/minogrpc/controller.TestMiniController_FailMarshalKey_OnStart(0xc0001056c0?)
        G:/projects/dedis/dela/mino/minogrpc/controller/mod_test.go:147 +0x207
testing.tRunner(0xc000105860, 0x125ed68)
        C:/Program Files/Go/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
        C:/Program Files/Go/src/testing/testing.go:1629 +0x3ea
```

--- a/internal/testing/fake/store.go
+++ b/internal/testing/fake/store.go
@@ -130,7 +130,13 @@ func (tx dbTx) GetBucket(name []byte) kv.Bucket {
 
 // GetBucketOrCreate implements kv.WritableTx.
 func (tx dbTx) GetBucketOrCreate(name []byte) (kv.Bucket, error) {
-	return tx.GetBucket(name), tx.err
+	bucket, found := tx.buckets[string(name)]
+	if !found {
+		bucket = NewBucket()
+		tx.buckets[string(name)] = bucket
+	}
+
+	return bucket, tx.err
 }
 
 // GetBucketOrCreate implements store.Transaction.

--- a/mino/minogrpc/controller/mod_test.go
+++ b/mino/minogrpc/controller/mod_test.go
@@ -436,15 +436,15 @@ func (b badCurve) IsOnCurve(x, y *big.Int) bool {
 }
 
 func (b badCurve) Add(x1, y1, x2, y2 *big.Int) (x, y *big.Int) {
-	panic("unsupported")
+	panic("not implemented - should not be needed by tests")
 }
 
 func (b badCurve) Double(x1, y1 *big.Int) (x, y *big.Int) {
-	panic("unsupported")
+	panic("not implemented - should not be needed by tests")
 }
 
 func (b badCurve) ScalarMult(x1, y1 *big.Int, k []byte) (x, y *big.Int) {
-	panic("unsupported")
+	panic("not implemented - should not be needed by tests")
 }
 
 func (b badCurve) ScalarBaseMult(k []byte) (x, y *big.Int) {

--- a/mino/minogrpc/controller/mod_test.go
+++ b/mino/minogrpc/controller/mod_test.go
@@ -2,6 +2,7 @@ package controller
 
 import (
 	"crypto/elliptic"
+	"math/big"
 	"net"
 	"os"
 	"path/filepath"
@@ -137,12 +138,15 @@ func TestMiniController_FailGenerateKey_OnStart(t *testing.T) {
 
 func TestMiniController_FailMarshalKey_OnStart(t *testing.T) {
 	ctrl := NewController().(miniController)
-	ctrl.curve = badCurve{Curve: elliptic.P224()}
+	ctrl.curve = badCurve{}
 
 	inj := node.NewInjector()
 	inj.Inject(fake.NewInMemoryDB())
 
-	str := map[string]string{"routing": "flat"}
+	str := map[string]string{
+		"routing": "flat",
+		"config":  "/THIS_PATH_DEFINITELY_DOESNT_EXIST/",
+	}
 
 	err := ctrl.OnStart(fakeContext{str: str}, inj)
 	require.EqualError(t, err, "failed to get cert option: cert private key: "+
@@ -418,6 +422,31 @@ func (badReader) Read([]byte) (int, error) {
 	return 0, fake.GetError()
 }
 
-type badCurve struct {
-	elliptic.Curve
+type badCurve struct{}
+
+func (b badCurve) Params() *elliptic.CurveParams {
+	return &elliptic.CurveParams{
+		Name: "BadCurve",
+		N:    big.NewInt(65536),
+	}
+}
+
+func (b badCurve) IsOnCurve(x, y *big.Int) bool {
+	return false
+}
+
+func (b badCurve) Add(x1, y1, x2, y2 *big.Int) (x, y *big.Int) {
+	panic("unsupported")
+}
+
+func (b badCurve) Double(x1, y1 *big.Int) (x, y *big.Int) {
+	panic("unsupported")
+}
+
+func (b badCurve) ScalarMult(x1, y1 *big.Int, k []byte) (x, y *big.Int) {
+	panic("unsupported")
+}
+
+func (b badCurve) ScalarBaseMult(k []byte) (x, y *big.Int) {
+	return big.NewInt(0), big.NewInt(0)
 }

